src/main/java/ru/yandex/clickhouse/ClickHousePreparedAbstractStatementImpl.java

```diff
@@ -0,0 +1,447 @@
+package ru.yandex.clickhouse;
+
+import org.apache.http.entity.AbstractHttpEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import ru.yandex.clickhouse.response.ClickHouseResponse;
+import ru.yandex.clickhouse.settings.ClickHouseProperties;
+import ru.yandex.clickhouse.settings.ClickHouseQueryParam;
+import ru.yandex.clickhouse.util.ClickHouseArrayUtil;
+import ru.yandex.clickhouse.util.guava.StreamUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.net.URL;
+import java.sql.*;
+import java.sql.Date;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+public abstract class ClickHousePreparedAbstractStatementImpl extends ClickHouseStatementImpl implements ClickHousePreparedStatement {
+    private final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+    private final SimpleDateFormat dateTimeFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+
+    public ClickHousePreparedAbstractStatementImpl(CloseableHttpClient client,
+                                                   ClickHouseConnection connection,
+                                                   ClickHouseProperties properties,
+                                                   TimeZone timezone) throws SQLException {
+
+        super(client, connection, properties);
+        initTimeZone(timezone);
+    }
+
+    private void initTimeZone(TimeZone timeZone) {
+        dateTimeFormat.setTimeZone(timeZone);
+        if (properties.isUseServerTimeZoneForDates())
+            dateFormat.setTimeZone(timeZone);
+    }
+
+
+    protected abstract void setParameter(int index, String value, boolean isQuote);
+
+    protected abstract String getSqlQuery() throws SQLException;
```

Comment:
I think buildSql() says more, because one may misread getSqlQuery as a simple getter while it isn't.
Maybe there is a better name but it must define that it is no simple getter.